streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStore.java

```diff
@@ -102,4 +139,232 @@ static class RawAndDeserializedValue<ValueType> {
             this.value = value;
         }
     }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <R> QueryResult<R> query(final Query<R> query,
+                                    final PositionBound positionBound,
+                                    final QueryConfig config) {
+
+        final long start = time.nanoseconds();
+        final QueryResult<R> result;
+
+        final StoreQueryUtils.QueryHandler handler = queryHandlers.get(query.getClass());
+        if (handler == null) {
+            result = wrapped().query(query, positionBound, config);
+            if (config.isCollectExecutionInfo()) {
+                result.addExecutionInfo(
+                        "Handled in " + getClass() + " in " + (time.nanoseconds() - start) + "ns");
+            }
+        } else {
+            result = (QueryResult<R>) handler.apply(
+                    query,
+                    positionBound,
+                    config,
+                    this
+            );
+            if (config.isCollectExecutionInfo()) {
+                result.addExecutionInfo(
+                        "Handled in " + getClass() + " with serdes "
+                                + serdes + " in " + (time.nanoseconds() - start) + "ns");
+            }
+        }
+        return result;
+    }
+
+
+
+    @SuppressWarnings("unchecked")
+    protected <R> QueryResult<R> runTimestampedKeyQuery(final Query<R> query,
```

Comment:
Why do we add this as `protected`? Seems it could be `private`? (same below)

Reply:
Because in `MeteredKeyValueStore`, we use protected one. So I follow the `MeteredKeyValueStore`'s one.

Reply:
Why is it `protected` in `MeteredKeyValueStore` -- I believe it should be `private` there, too? Can you try it out?

Reply:
I don't know why the contributor use protected here, but I will try it out.

Reply:
Hi, @mjsax, if we change `protected` to `private`, it has a related problem, that is we have to change all other class's `runRangeQuery` or `runKeyQuery` methods from `proteced` to `private` and remove `@Override `annotation, such as the `runRangeQuery` method in `MeteredVersionKeyValueStore`.